src/gui/dialogs/multiplayer/mp_join_game.cpp

```diff
@@ -218,15 +219,15 @@ static std::string generate_user_description(const config& side)
 	const std::string reservation = side["current_player"].str();
 	const std::string owner = side["player_id"].str();
 
-	if(controller_type == "ai") {
+	if(controller_type == side_controller::ai) {
 		return _("Computer Player");
-	} else if(controller_type == "null") {
+	} else if(controller_type == side_controller::none) {
 		return _("Empty slot");
-	} else if(controller_type == "reserved") {
+	} else if(controller_type == side_controller::reserved) {
 		return VGETTEXT("Reserved for $playername", {{"playername", reservation}});
 	} else if(owner.empty()) {
 		return _("Vacant slot");
-	} else if(controller_type == "human" || controller_type == "network") {
+	} else if(controller_type == side_controller::human) {
```

Comment:
Both the old and the new code feel wrong - the MAKE_ENUM framework supports converting to and from strings, so why isn't this using it?

Reply:
I don't know why a variety of places aren't using the CONTROLLER MAKE_ENUM in places like these. Also note that the wesnothd and client MAKE_ENUMs aren't even common code - they're copy/pastes of the same few lines into two separate headers.